src/file_scanner/arrow_multi_file_info.cpp

```diff
@@ -0,0 +1,244 @@
+#include "file_scanner/arrow_multi_file_info.hpp"
+
+#include "ipc/stream_reader/ipc_file_stream_reader.hpp"
+
+#include "duckdb/common/bind_helpers.hpp"
+#include "file_scanner/arrow_file_scan.hpp"
+#include "ipc/stream_factory.hpp"
+
+namespace duckdb {
+namespace ext_nanoarrow {
+
+unique_ptr<BaseFileReaderOptions> ArrowMultiFileInfo::InitializeOptions(
+    ClientContext& context, optional_ptr<TableFunctionInfo> info) {
+  return make_uniq<ArrowFileReaderOptions>();
+}
+
+bool ArrowMultiFileInfo::ParseCopyOption(ClientContext& context, const string& key,
+                                         const vector<Value>& values,
+                                         BaseFileReaderOptions& options_p,
+                                         vector<string>& expected_names,
+                                         vector<LogicalType>& expected_types) {
+  return true;
+}
+
+bool ArrowMultiFileInfo::ParseOption(ClientContext& context, const string& key,
+                                     const Value& val, MultiFileOptions& file_options,
+                                     BaseFileReaderOptions& options) {
+  return true;
+}
+
+void ArrowMultiFileInfo::FinalizeCopyBind(ClientContext& context,
+                                          BaseFileReaderOptions& options_p,
+                                          const vector<string>& expected_names,
+                                          const vector<LogicalType>& expected_types) {}
+
+struct ArrowMultifileData : public TableFunctionData {
```

Comment:
nit: `ArrowMultiFileData` to adhere to the other names? (note the capitalization of File)